src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1791,6 +1818,72 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                             INS_SCALABLE_OPTS_UNPREDICATED);
                 break;
 
+            case NI_Sve_SaturatingDecrementBy16BitElementCountScalar:
+            case NI_Sve_SaturatingDecrementBy32BitElementCountScalar:
+            case NI_Sve_SaturatingDecrementBy64BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy16BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy32BitElementCountScalar:
+            case NI_Sve_SaturatingIncrementBy64BitElementCountScalar:
+                // Use scalar sizes.
+                emitSize = emitActualTypeSize(node->gtType);
+                opt      = INS_OPTS_NONE;
+                FALLTHROUGH;
+
+            case NI_Sve_SaturatingDecrementBy16BitElementCount:
+            case NI_Sve_SaturatingDecrementBy32BitElementCount:
+            case NI_Sve_SaturatingDecrementBy64BitElementCount:
+            case NI_Sve_SaturatingDecrementBy8BitElementCount:
+            case NI_Sve_SaturatingIncrementBy16BitElementCount:
+            case NI_Sve_SaturatingIncrementBy32BitElementCount:
+            case NI_Sve_SaturatingIncrementBy64BitElementCount:
+            case NI_Sve_SaturatingIncrementBy8BitElementCount:
+            {
+                assert(isRMW);
+                if (targetReg != op1Reg)
+                {
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
+                }
+
+                if (intrin.op2->IsCnsIntOrI() && intrin.op3->IsCnsIntOrI())
+                {
+                    // Both immediates are constant, emit the intruction.
+
+                    assert(intrin.op2->isContainedIntOrIImmed() && intrin.op3->isContainedIntOrIImmed());
+                    int           scale   = (int)intrin.op2->AsIntCon()->gtIconVal;
+                    insSvePattern pattern = (insSvePattern)intrin.op3->AsIntCon()->gtIconVal;
+                    GetEmitter()->emitIns_R_PATTERN_I(ins, emitSize, targetReg, pattern, scale, opt);
+                }
+                else
+                {
+                    // Use the helper to generate a table.
+
+                    assert(!intrin.op2->isContainedIntOrIImmed() && !intrin.op3->isContainedIntOrIImmed());
+
+                    emitAttr scalarSize = emitActualTypeSize(node->GetSimdBaseType());
+
+                    // Combine the second immediate (pattern, op3) into the first (scale, op2).
+                    GetEmitter()->emitIns_R_R_I(INS_sub, scalarSize, op2Reg, op2Reg, 1);
```

Comment:
can you add a pseudocode of assembly instruction of what we are trying to do for future reference?

Reply:
To start with, a comment about `op2Reg` has `scale` and `op3Reg` has pattern, etc. and the a comment for each operation.